### ISSUE TITLE
Input, Input Message - Provide a clear indication of fields in error for information that is submitted - (2036569949)

### ISSUE THREAD

**Issue by dqateam** [Esri/calcite-design-system#7792]
### Violation: 
 Provide a clear indication of fields in error for information that is submitted 

### WCAG Reference: 
- [3.3.1 Error Identification](https://www.w3.org/WAI/WCAG21/Understanding/error-identification) 

### Severity: 
 6 

### Media Type: 
 Forms 

--- 
### Areas for Remediation: 
- Title: Input, Input Message - Provide a clear indication of fields in error for information that is submitted - (2036569949) 
- Module: Input, Input Message 
**Issue**
There are errors/ informative messages that do not clearly describe the messages. Examples include:

- Success message
- warning message
- error message

**User Impact**
Screen reader users and users with cognitive disabilities will have difficulty determining that these messages are errors instead of general instructions on how to fill the fields. They will also have difficulty determining which error corresponds to which field.

**Code Reference** 
 ```
N/A
 ``` 
**Suggestion**
Provide a clear indication of fields in error for information that is submitted. Include the name of the field in the error message. In case of any informative message, the message should be clear enough.

**Compliant Code Example** 
 ```
N/A
 ```
 
--- 
### Additional Resources: 
- Report Source: Product accessibility evaluation conducted on the Calcite Design System in August, 2022.

**Comment by driskull** [comment]
Seems like we can address this as part of the form validation epic we have. I would say this isn't valid since we haven't tacked that effort yet.

**Comment by geospatialem** [comment]
Blocking the above issue until the efforts from https://github.com/Esri/calcite-design-system/issues/4598 are incorporated in across `input`-named components, anticipated for Jan/Feb 2024.

**Comment by geospatialem** [comment]
@tyanthony Thanks for re-pinging this issue, we're winding down the efforts of our form validation effort [mentioned above](https://github.com/Esri/calcite-design-system/issues/7792#issuecomment-1870639292), and this still seems to be a valid bug. Will look at addressing in a future milestone in the next few weeks:

---

Current unsupported example with `2.8.0`: https://codepen.io/geospatialem/pen/jORoLWq

---

_Future implementation note:_
We should provide the `validationMessage` (default or user-defined) to AT by:
- Adding an `aria-invalid` to the component when `status="invalid"` and 
- Associating the `validationMessage` contents using `aria-errorMessage`

For instance: 
- Native HTML: https://codepen.io/geospatialem/pen/MWdYMWJ and 
- CC workaround/implementation: https://codepen.io/geospatialem/pen/XWwJLjV

**Comment by driskull** [comment]
I don't think we need to do anything to `input-message` since it is primarily being used internally now.

**Comment by geospatialem** [comment]
Verified with JAWS and NVDA on Windows 🏆
